staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go

```diff
@@ -633,6 +633,13 @@ func TestDeltaFIFO_HasSynced(t *testing.T) {
 			},
 			expectedSynced: true,
 		},
+		{
+			actions: []func(f *DeltaFIFO){
+				func(f *DeltaFIFO) { f.Replace([]interface{}{mkFifoObj("a", 1), mkFifoObj("a", 2)}, "0") },
```

Comment:
Replace always captures a complete snapshot consistent in time; there cannot be duplicate keys in the list or apiserver is broken.

...I guess maybe it can still be tested but please add a comment explaining that (and why) this is not testing a normal codepath.

Reply:
Thanks. Added comment per suggestion. @lavalamp

Reply:
The wording from @lavalamp , copied into the comment, seems a bit misdirected to me.  Rather than saying "Replace always captures" --- which suggests that the responsibility for this behavior is inside Replace --- I suggest something more like "A Reflector always passes to Replace".  It looks to me like the whole point of this PR is to move beyond the limited usage documented at https://github.com/kubernetes/kubernetes/blob/e3de62298a730415c5d2ab72607ef6adadd6304d/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go#L134-L135 ("where a Reflector is intended to be the producer").

Reply:
Yeah perhaps "expects" would have been a better word than "captures".

Having the comment start on the same line as the brace looks a little odd, I don't think we do that usually?

If you can fix these two tiny nits I can LGTM

Reply:
Thanks @MikeSpreitzer and @lavalamp. 

I rephrased the comments in the test to 

> This test case won't happen in practice since a Reflector, the only producer for delta_fifo today, always passes a complete snapshot consistent in time; there cannot be duplicate keys in the list or apiserver is broken.

Please let me know if this looks better.